46.md

```diff
@@ -1,98 +1,162 @@
+To initiate a connection between a client and a remote signer there are a few different options.
 
-### Started by the signer (nsecBunker)
+### Direct connection initiated by remote signer
 
-The remote signer generates a connection token in the form
+This is most common in a situation where you have your own nsecbunker or other type of remote signer and want to connect through a client that supports remote signing.
+
+The remote signer would provide a connection token in the form:
 
 ```
-bunker://<hex-pubkey>?relay=wss://...&relay=wss://...&secret=<optional-secret>
+bunker://<remote-pubkey>?relay=<wss://relay-to-connect-on>&relay=<wss://another-relay-to-connect-on>&secret=<optional-secret-value>
 ```
 
-The user copies that token and pastes it in the client UI somehow. Then the client can send events of kind `24133` to the specified relays and wait for responses from the remote signer.
+This token is pasted into the client by the user and the client then uses the details to connect to the remote signer via the specified relay(s).
 
-### Started by the client
+### Direct connection initiated by the client
 
-The client generates a QR code in the following form (URL-encoded):
+In this case, basically the opposite direction of the first case, the client provides a connection token (or encodes the token in a QR code) and the signer initiates a connection to the client via the specified relay(s).
 
 ```
-nostrconnect://<client-key-hex>?relay=wss://...&metadata={"name":"...", "url": "...", "description": "..."}
+nostrconnect://<local-keypair-pubkey>?relay=<wss://relay-to-connect-on>&metadata=<json metadata in the form: {"name":"...", "url": "...", "description": "..."}>
 ```
 
-The signer scans the QR code and sends a `connect` message to the client in the specified relays.
+### Remote signer discovery via NIP-89
+
+In this last case, most often used to fascilitate an OAuth-like signin flow, the client first looks for remote signers that have announced themselves via NIP-89 application handler events.
+
+First the client will query for `kind: 31989` events that have a `k` tag of `24133`.
+
+These are generally shown to a user, and once the user selects which remote signer to use and provides the remote pubkey they want to use (via npub, pubkey, or nip-05 value), the client can initiate a connection. Note that it's on the user to select the remote signer that is actually managing the remote key that they would like to use in this case. If the remote pubkey is managed on another remote signer, the connection will fail.
+
+In addition, it's important that clients validate that the pubkey of the announced remote signer matches the pubkey of the `_` entry in the `/.well-known/nostr.json` file of the remote signer's announced domain.
```

Comment:
What domain is this? Right now it's inferred from the nip05 in the kind 31990 event, but I think some new tags should be introduced, maybe `signer_domain`, `signer_relays`. Domain would be optional to support the non-publicly-accessible use case. `signer_relays` would replace the `nip46` field in `nostr.json`. This way, DNS is not a hard dependency for discoverable signers. If a domain is provided, you'd still want to validate that the domain's pubkey matches the author's pubkey.

Reply:
Should that get added here or in NIP-89?